tests/hikari/impl/test_rest.py

```diff
@@ -292,6 +292,199 @@ def test_invalidate_when_token_is_stored_token(self):
+        )
+
+    def test_client_id_property(self):
+        mock_client = hikari_test_helpers.mock_class_namespace(applications.Application, id=41551, init_=False)()
+        strategy = rest.OAuthCredentialsStrategy(client=mock_client, client_secret="123123123", auth_code="auth#code", redirect_uri="https://web.site/auth/discord")
+
+        assert strategy.client_id == 41551
+
+    def test_scopes_property(self):
+        token = rest.OAuthCredentialsStrategy(client=987654321, client_secret="123123123", auth_code="auth#code", redirect_uri="https://web.site/auth/discord", scopes=("identify",))
+
+        assert token.scopes == ("identify",)
+
+    def test_token_type_property(self):
+        token = rest.OAuthCredentialsStrategy(client=987654321, client_secret="123123123", auth_code="auth#code", redirect_uri="https://web.site/auth/discord")
+        assert token.token_type is applications.TokenType.BEARER
+
+    @pytest.mark.asyncio()
+    async def test_acquire_on_new_instance(self, mock_token):
+        mock_rest = mock.AsyncMock(authorize_access_token=mock.AsyncMock(return_value=mock_token))
+
+        result = await rest.OAuthCredentialsStrategy(client=987654321, client_secret="123123123", auth_code="auth#code", redirect_uri="https://web.site/auth/discord", scopes=("identify",)).acquire(mock_rest)
+
+        assert result == "mockmock.tokentoken.mocktoken"
+
+        mock_rest.authorize_access_token.assert_awaited_once_with(
+            client=987654321,
+            client_secret="123123123",
+            code="auth#code",
+            redirect_uri="https://web.site/auth/discord",
+        )
+
+    @pytest.mark.asyncio()
+    async def test_acquire_handles_out_of_date_token(self, mock_token):
+        mock_old_token = mock.AsyncMock(
+            applications.PartialOAuth2Token,
+            expires_in=datetime.timedelta(weeks=1),
+            token_type=applications.TokenType.BEARER,
+            access_token="old.mock.token",
+            refresh_token="refresh.mock.token"
+        )
+        mock_rest = mock.AsyncMock(refresh_access_token=mock.AsyncMock(return_value=mock_token))
+        strategy = rest.OAuthCredentialsStrategy(client=123456789, client_secret="123123123", auth_code="auth#code", redirect_uri="https://web.site/auth/discord")
+        token = await strategy.acquire(
+            mock.AsyncMock(authorize_access_token=mock.AsyncMock(return_value=mock_old_token))
+        )
+
+        with mock.patch.object(time, "monotonic", return_value=99999999999):
+            new_token = await strategy.acquire(mock_rest)
+
+        mock_rest.refresh_access_token.assert_awaited_once_with(
+            client=123456789, client_secret="123123123", refresh_token="refresh.mock.token"
+        )
+        assert new_token != token
+        assert new_token == "mockmock.tokentoken.mocktoken"
+
+    @pytest.mark.asyncio()
+    async def test_acquire_handles_token_being_set_before_lock_is_acquired(self, mock_token):
+        lock = asyncio.Lock()
```

Comment:
Why the use of a lock here?

Reply:
Followed tests from ClientCredentialsStrategy

Reply:
Ah, ok. Got why thats there now. Its to prevent the gatherer from attempting to go too quickly. Dont think its necessary in this case, but its anyones guess. Ill do some tests and see if it changes anything

Reply:
Running the test without locking seems fine.  Ran multiple times and no errors